### PR TITLE
chore: switch to the new debug controller harness

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/cli/cli.ts [hunk 1]
```diff
@@ -682,8 +682,8 @@ function logErrorAndExit(e: Error) {
   process.exit(1);
 }
 
-function language(): string {
-  return process.env.PW_LANG_NAME || 'test';
+function codegenId(): string {
+  return process.env.PW_LANG_NAME || 'javascript-test';
```
Comment (mxschmitt) [inline packages/playwright-core/src/cli/cli.ts:686]: isn't it by definition more TS? Or did we make "import" work in JS well?

**Review by mxschmitt — APPROVED** [review]

File: packages/playwright-core/src/cli/driver.ts [hunk 2]
```diff
@@ -55,8 +52,6 @@ export async function runServer(port: number | undefined, path = '/', maxConnect
   process.on('exit', () => server.close().catch(console.error));
   console.log('Listening on ' + wsEndpoint);  // eslint-disable-line no-console
   process.stdin.on('close', () => selfDestruct());
-  if (reuseBrowser && process.send)
```
Comment (dgozman) [inline packages/playwright-core/src/cli/driver.ts:58]: `reuseBrowser` is unused now?

File: tests/library/debug-controller.spec.ts [hunk 3]
```diff
@@ -153,3 +143,33 @@ test('should highlight all', async ({ backend, connectedBrowser }) => {
...
+  backend.on('sourceChanged', event => events.push(event));
+
+  await backend.setMode({ mode: 'recording' });
+
+  const context = await connectedBrowser._newContextForReuse();
+  const [page] = context.pages();
+
+  await page.locator('body').click();
+
+  await new Promise(f => setTimeout(f, 2000));
```
Comment (dgozman) [inline tests/library/debug-controller.spec.ts:158]: Don't need this one probably?

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/cli/cli.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/cli/cli.ts:686]: Hm, I did not think it about it. import does work in JS, but I wonder if we should differentiate between JS and TS now...

File: tests/library/debug-controller.spec.ts (hunk 3 above)
Comment (pavelfeldman) [reply tests/library/debug-controller.spec.ts:158]: Oh yes.

File: packages/playwright-core/src/cli/cli.ts (hunk 1 above)
Comment (mxschmitt) [reply packages/playwright-core/src/cli/cli.ts:685]: tested with js/ts/js-esm all good, lgtm.

**Comment by dtinth** [comment]
This changes breaks `playwright-ruby-client` as it is sending `sdkLanguage: 'ruby'`

- https://github.com/YusukeIwaki/playwright-ruby-client/issues/228

Maybe the schema should allow experimental languages (perhaps identified by some prefix like `x-`)? Otherwise the alternative is that 3rd party clients have to masquerade as one of (javascript|python|java|csharp).

**Pull request merged**